### PR TITLE
ENH allow to delete a user from database

### PR DESCRIPTION
**Pull request by glemaitre**
#### Reference Issue

partially addressing #270

### CODE REVIEW

File: ramp-database/ramp_database/tools/user.py [hunk 1]
```diff
@@ -104,6 +104,21 @@ def add_user(session, name, password, lastname, firstname, email,
...
+
+    Parameters
+    ----------
+    session : :class:`sqlalchemy.orm.Session`
+        The session to directly perform the operation on the database.
+    name : str
+        The name of the user.
+    """
+    user = session.query(User).filter(User.name == name).one()
+    session.delete(user)
```
Comment (agramfort) [inline ramp-database/ramp_database/tools/user.py]: Should we delete also teams he belonged to and the submissions ?

Comment (glemaitre) [reply ramp-database/ramp_database/tools/user.py]: `session.delete` will make a cascade delete where there is a foreign key. Team should have a foreign key there. What could be added is that it really happen in the team table.

Comment (glemaitre) [reply ramp-database/ramp_database/tools/user.py]: I pushed a quick additional check: https://github.com/paris-saclay-cds/ramp-board/commit/c73e2fa3d7f6bf15a88e7b72dc45aeb8e2cac69c

**Comment by agramfort** [comment]
great

corresponding submissions are deleted too?

when you delete a submission you remove files from disk?
bagged score should then technically be recomputed if this submission was
present
but it's for another iteration. Just raising the issue

>

**Comment by glemaitre** [comment]
> corresponding submissions are deleted too?

If they have a foreign key link to the user, they will be.

> when you delete a submission you remove files from disk?

At least from the database

>bagged score should then technically be recomputed if this submission was
present but it's for another iteration. Just raising the issue

True

**Pull request merged**